pkg/polaris/polaris.go

```diff
@@ -46,57 +44,75 @@ const (
 	DefaultServiceAccountFile = "~/.rubrik/polaris-service-account.json"
 )
 
-// Client is used to make calls to the Polaris platform.
+// Client is used to make calls to the RSC platform.
 type Client struct {
 	GQL *graphql.Client
-	Log log.Logger
+	Log log.Logger // Deprecated: use the logger passed in to NewClientWithLogger.
 }
 
-// Deprecated: use aws.NewAPI(client.GQL) instead.
-func (c *Client) AWS() aws.API {
-	return aws.NewAPI(c.GQL)
+// Account represents a Polaris account. Implemented by UserAccount and
+// ServiceAccount.
+type Account interface {
+	isAccount()
 }
 
-// Deprecated: use azure.NewAPI(client.GQL) instead.
-func (c *Client) Azure() azure.API {
-	return azure.NewAPI(c.GQL)
+// Deprecated: The context and the logger parameters will be dropped in the next
+// release, use NewClientWithLogger
+func NewClient(ctx context.Context, account Account, logger log.Logger) (*Client, error) {
+	return NewClientWithLogger(account, logger)
 }
 
-// Deprecated: use gcp.NewAPI(client.GQL) instead.
-func (c *Client) GCP() gcp.API {
-	return gcp.NewAPI(c.GQL)
+// NewClientWithLogger returns a new Client for the specified Account.
+func NewClientWithLogger(account Account, logger log.Logger) (*Client, error) {
+	var client *Client
+
+	var err error
+	switch account := account.(type) {
+	case *UserAccount:
+		client, err = newClientFromUserAccount(account, logger)
+	case *ServiceAccount:
+		client, err = newClientFromServiceAccount(account, logger)
+	default:
+		err = errors.New("invalid account type")
+	}
+	if err != nil {
+		return nil, fmt.Errorf("failed to create client: %v", err)
+	}
+
+	return client, nil
 }
 
-// Account represents a Polaris account. Implemented by UserAccount and
-// ServiceAccount.
-type Account interface {
-	isAccount()
+// SetLogger sets the logger to use.
+func (c *Client) SetLogger(logger log.Logger) {
+	c.GQL.SetLogger(logger)
 }
 
-// NewClient returns a new Client from the specified Account. The log level of
-// the given logger can be changed at runtime using the environment variable
-// RUBRIK_POLARIS_LOGLEVEL.
-func NewClient(ctx context.Context, account Account, logger log.Logger) (*Client, error) {
-	if serviceAccount, ok := account.(*ServiceAccount); ok {
-		return newClientFromServiceAccount(ctx, serviceAccount, logger)
+// LogLevelFromEnv sets the log level of the logger to the log level specified
+// in the RUBRIK_POLARIS_LOGLEVEL environment variable.
+func LogLevelFromEnv(logger log.Logger) error {
```

Comment:
nits: Should we include `Set` in the name to slightly clarify what it does? `SetLogLevelFromEnv(..)`